### PR TITLE
feat(forms/Datalist): Adds remote suggestions fetching

### PR DESCRIPTION
**Pull request by talend-glorieux**
**What is the problem this PR is trying to solve?**
Datalist doesn't allow remote suggestions lists.

**What is the chosen solution to this problem?**
* Adds remote suggestions fetching
* Fixes up/down navigation
* BREAKING CHANGES: The input isn't exclusive to the suggestions list

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/src/widgets/DatalistWidget/DatalistWidget.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ export function escapeRegexCharacters(str) {
  * @param value
  */
 export function getMatchingSuggestions(suggestions, value) {
+	if (!suggestions) {
```
Comment (jsomsanith) [inline packages/forms/src/widgets/DatalistWidget/DatalistWidget.js]: Prefer default value

File: packages/forms/src/widgets/DatalistWidget/DatalistWidget.js [hunk 2]
```diff
@@ -66,7 +69,7 @@ function renderDatalistItem(item, { value }) {
 			emphasisedText.push(restValues[i]);
 			if (matchedValues[i]) {
 				emphasisedText.push(
-					<em className={theme['highlight-match']}>{matchedValues[i]}</em>
+					<em className={theme['highlight-match']}>{matchedValues[i]}</em>,
```
Comment (jsomsanith) [inline packages/forms/src/widgets/DatalistWidget/DatalistWidget.js]: Remove that comma please

File: packages/forms/src/widgets/DatalistWidget/DatalistWidget.js [hunk 3]
```diff
@@ -95,10 +98,30 @@ function renderNoMatch() {
...
+			enum: PropTypes.arrayOf(PropTypes.string),
+		}).isRequired,
+		formContext: PropTypes.shape({
+			fetchItems: PropTypes.func,
+		}),
+		options: PropTypes.shape({
+			// Is the field value restricted to the suggestion list
+			restricted: PropTypes.bool,
+		}),
+	}
```
Comment (jsomsanith) [inline packages/forms/src/widgets/DatalistWidget/DatalistWidget.js]: Add a semicolon please

File: packages/forms/src/widgets/DatalistWidget/DatalistWidget.js [hunk 4]
```diff
@@ -168,7 +194,27 @@ class DatalistWidget extends React.Component {
 	}
 
 	initSuggestions(value) {
-		const suggestions = getMatchingSuggestions(this.props.schema.enum, value);
+		let items;
+		if (this.props.schema.enum) {
+			items = this.props.schema.enum;
+		} else if (this.props.formContext && this.props.formContext.fetchItems) {
+			items = this.props.formContext.fetchItems(this.props.schema.title);
```
Comment (jsomsanith) [inline packages/forms/src/widgets/DatalistWidget/DatalistWidget.js]: Something based on the title :(
But right now we don't have anything else, so I understand. That may change with next version of forms.

File: packages/forms/src/widgets/DatalistWidget/DatalistWidget.scss [hunk 5]
```diff
@@ -34,6 +34,7 @@ $tf-typeahead-no-result-color: $silver !default;
 	border-radius: $border-radius-base;
 	box-shadow: 0 1px 3px 0 $tf-typeahead-items-shadow-color;
 	overflow-y: auto;
+	z-index: 999;
```
Comment (jsomsanith) [inline packages/forms/src/widgets/DatalistWidget/DatalistWidget.scss]: Why ?

File: packages/forms/stories/index.js [hunk 6]
```diff
@@ -114,6 +114,82 @@ decoratedStories.add('Multiple actions', () => {
 	);
 });
 
+decoratedStories.add('Datalist', () => {
+	function fetchItems(title) {
+		console.log('fetchItems', title);
+		return [
```
Comment (jsomsanith) [inline packages/forms/stories/index.js]: Can you change the content to show in the story that the list is different from the enum in jsonSchema ?

File: packages/forms/src/widgets/DatalistWidget/DatalistWidget.js (hunk 4 above)
Comment (talend-glorieux) [inline packages/forms/src/widgets/DatalistWidget/DatalistWidget.js]: Yep I guess we'll be able to pass things like URLs later.

File: packages/forms/src/widgets/DatalistWidget/DatalistWidget.scss (hunk 5 above)
Comment (talend-glorieux) [inline packages/forms/src/widgets/DatalistWidget/DatalistWidget.scss]: Always on top. It will break on Drawers if not added.

Comment (jsomsanith) [inline packages/forms/src/widgets/DatalistWidget/DatalistWidget.scss]: Oh sh...

**Pull request merged**